src/lj_jitlog.c

```diff
@@ -225,31 +225,42 @@ static void jitlog_tracestart(JITLogState *context, GCtrace *T)
   context->lastpc = proto_bcpos(J->pt, J->pc);
 }
 
-static void jitlog_tracestop(JITLogState *context, GCtrace *T)
+static void jitlog_writetrace(JITLogState *context, GCtrace *T, int abort)
 {
-  jit_State *J = G2J(context->g);
+  jit_State *J = G2J(context->g);  
   GCproto *startpt = &gcref(T->startpt)->pt;
+  BCPos startpc = proto_bcpos(startpt, mref(T->startpc, const BCIns));
   BCPos stoppc;
   GCproto *stoppt = getcurlualoc(context, &stoppc);
-  BCPos startpc = proto_bcpos(startpt, mref(T->startpc, const BCIns));
   memorize_proto(context, startpt);
   memorize_proto(context, stoppt);
   memorize_func(context, context->lastfunc);
-  log_trace(context->g, T, 0, J->parent, stoppt, stoppc, context->lastfunc, 0, startpc, T->ir + T->nk, (T->nins - T->nk) + 1);
+
+  int abortreason = (abort && tvisnumber(J->L->top-1)) ? numberVint(J->L->top-1) : -1;
+  int irsize = (T->nins - T->nk) + 1;
+  MSize mcodesize = T->szmcode;
+  if (context->user.logfilter & LOGFILTER_TRACE_MCODE)
```

Comment:
These two `if`s would be clearer (and a teeny tiny bit faster) as:

```c
int z;
if (x) z = 0; else z = 1;
```
rather than:

```c
int z = 0;
if (x) z = 1;
```

Reply:
i just went with the least verbose way

Reply:
I realise what you did ;) I'm suggesting you change it to something clearer.

Reply:
i don't know if there is anything clearer 
```
int irsize = (context->user.logfilter & LOGFILTER_TRACE_IR) ? (T->nins - T->nk) + 1 : 0;
```
would be worse and
```
if (context->user.logfilter & LOGFILTER_TRACE_IR) irsize = 0; else irsize = T->nins - T->nk; 
```
I  think just makes its much harder to read

Reply:
@ltratt ping

Reply:
I outlined the fix. The problem with the idiom you're using is that, while it's shorter, it assigns to the variable twice: that gives the reader (e.g. me) the impression that there's a semantic reason behind the double assignment (when there isn't).

Reply:
Its meant show the default is overridden when the flag is set

Reply:
That explanation doesn't make any sense when the default is overridden immediately.

Reply:
well i changed but i think its for the worse and travis is failing because it doesn't bother caching the clang packages

Reply:
I understand that you're not keen, but this is easier to read for me.

I don't think changing this change is causing the Travis failures, so that's best discussed elsewhere ;)